app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,21 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, general info, callstack
+    def self.deprecation_warning(foreman_version_deadline, options = {})
+      raise Foreman::Exception, _("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A\d[.]\d+[.]\d\z/)
+      controller = options[:controller]
```

Comment:
i added them to make the code more readable, if you don't agree Ill take them off?